scripts/sort_contibutors/main.py

```diff
@@ -0,0 +1,217 @@
+
+        # Filter the github bots
+        if '[bot]' not in username and username not in black_list:
+            contributors.append(
+                {'username': username, 'name': name, 'ranking': ranking})
+
+    return contributors
+
+# Retrieves the list of fullnames of contributors of a repository in JSON format
+
+
+def get_fullname(username: str, user_token: str) -> str:
+    headers = {'X-GitHub-Api-Version': '2022-11-28',
+               'Accept': 'application/vnd.github+json',
+               'Authorization': 'Bearer ' + user_token}
+    r = requests.get('https://api.github.com/users/' +
+                     username, headers=headers, timeout=20)
+    if r.status_code == 401:
+        print("Invalid token")
+        os._exit(-1)
+    return r.json()['name']
+
+# Retrieves the list of contributors of a repository in JSON format
+
+
+def fetch_repository(project: str, user_token: str) -> list:
+    headers = {'X-GitHub-Api-Version': '2022-11-28',
+               'Accept': 'application/vnd.github+json',
+               'Authorization': 'Bearer ' + user_token}
+    r = requests.get('https://api.github.com/repos/OWASP/' +
+                     project+'/contributors', headers=headers, timeout=20)
+    if r.status_code == 401:
+        print("Invalid token")
+        os._exit(-1)
+    return parse_contributor_list(r.json(), token)
+
+
+def merge_users(l: list) -> list:
+    username = dict()
+    ranking = dict()
+
+    for a in l:
+
+        if a['username'] in ranking:
+            ranking[a['username']] += a['ranking']
+        else:
+            ranking[a['username']] = a['ranking']
+        username[a['username']] = {'username': a['username'],
+                                   'name': a['name'], 'ranking': ranking[a['username']]}
+
+    l = dict(sorted(username.items(),
+             key=lambda x: x[1]['ranking'], reverse=True))
+
+    special_contributors = []
+    contributors = []
+
+    for key, value in l.items():
+        element = {'name': value['name'],
+                   'username': key, 'ranking': value['ranking']}
+        if element['ranking'] >= 100:
+            special_contributors.append(element)
+        else:
+            contributors.append(element)
+    return [special_contributors, contributors]
+
+
+def get_contibutors_list(token: str) -> list:
+
+    print("[+] Fetching the Wrong Secrets CTF party contributors list ... ")
+    wrongsecrets_ctf_list = fetch_repository('wrongsecrets-ctf-party', token)
+    print("[+] Fetching the Wrong Secrets Binaries contributors list ... ")
+    wrongsecrets_binaries_list = fetch_repository(
+        'wrongsecrets-binaries', token)
+    print("[+] Fetching the Wrong Secrets contributors list ... ")
+    wrongsecrets_list = fetch_repository('wrongsecrets', token)
+    print("[+] Fetching the Wrong Secrets contributors list ... ")
+    wrongsecrets_list = fetch_repository('wrongsecrets', token)
```

Comment:
is there a reason for fetching it twice?

Reply:
Oh, I'll fix it now .

Reply:
Sorry about that.

Reply:
It's fixed.